index.bs

```diff
@@ -0,0 +1,221 @@
+    type: dfn; url: #box-fragment; text: box fragment; 
+
+</pre>
+
+Introduction {#sec-intro}
+=====================
+
+<div class="non-normative">
+
+<em>This section is non-normative.</em>
+
+When the layout of a page is unstable it leads to a very bad user experience. This shifting of DOM elements is often due to content loading asynchronously and displacing other elements on the page. The layout instability metric identifies these unstable pages by measuring the layout shift fraction of each frame on the page, allowing the user to compute an overall instability score for the page.
+
+</div>
+
+Usage Example {#example}
+------------------------
+
+<pre class="example highlight">
+    var observer = new PerformanceObserver(function(list) {
+        var perfEntries = list.getEntries();
+        for (var i = 0; i < perfEntries.length; i++) {
+            // Process entries
+            // report back for analytics and monitoring
+            // ...
+        }
+    });
+
+    // register observer for layout shift notifications
+    observer.observe({entryTypes: ["layoutShift"]});
+</pre>
+
+Terminology {#sec-terminology}
+==============================
+An <a>element</a>’s <dfn export>starting point</dfn> refers to the <a>element</a>'s <a>flow-relative offset</a> in the document.
+
+The <dfn export>visual representation</dfn> of a <a href="https://www.w3.org/TR/html401/struct/global.html#h-7.5.3">block-level</a> <a>element</a> is its <a>border box</a>. The <a>visual representation</a> of an <a href="https://www.w3.org/TR/html401/struct/global.html#h-7.5.3">inline element</a> is the geometric union of its <a>box fragment</a>s, the first of which is used to determine its <a>starting point</a>. 
+
+
+
+
+{{LayoutShift}} interface {#sec-layout-shift}
+=======================================
+
+<pre class="idl">
+    interface LayoutShift : PerformanceEntry {
+      readonly attribute long value;
+    };
+</pre>
+
+{{LayoutShift}} extends the following attributes of {{PerformanceEntry}} interface:
+
+* The {{PerformanceEntry/name}} attribute must return the {{DOMString}} <code>"layout-shift"</code>
+* The {{PerformanceEntry/entryType}} attribute must return <code>"layoutShift"</code>.
+* The {{PerformanceEntry/startTime}} attribute must return a {{DOMHighResTimeStamp}} of when the shift occured.
+* The {{PerformanceEntry/duration}} attribute must return 0.
+* The <dfn attribute for=LayoutShift>value</dfn> attribute's getter will return the value it is set to.
+
+<dfn attribute for=PerformanceLongTaskTiming>attribution</dfn>
+A user agent implementing {{LayoutShift}} must perform the following steps:
+1. Run the <a>register a performance entry type</a> algorithm with <code>"layoustShift"</code> as input.
+
+
+NOTE: The <a>getEntriesByType</a> method will not return entries of this type.  To obtain instability events that occur before a <a>PerformanceObserver</a> has been registered, register the observer with <a>buffered</a> set to true.
```

Comment:
Does this need to be spec'ed? I see in the `performance-timeline` spec there's an [`availableFromTimeline`](https://w3c.github.io/performance-timeline/#performance-timeline) flag, but I actually don't see it mentioned in other specs (e.g. the [`resource-timing` spec](https://w3c.github.io/resource-timing) or the [`paint-timing` spec](https://w3c.github.io/paint-timing/)), so it's not clear to me how a UA implementing this would know what to do with all these entry types.

Reply:
Right, so availableFromTimeline is a very new variable in the buffered flag spec.  We will be updating the spec shortly to add initial values for these variables and for all new specs it will be set to FALSE.  

I agree that this is not necessary here.  The "register a performance entry type" algorithm will probably be changing soon at which point we will update this spec as well.

Reply:
availableFromTimeline was a very recent addition and we're going to have the values set in a separate "WG Note" that is linked from PerformanceTimeline. Also worth noting that this was just in a Note, which means it was there as non-normative text.